tests/test_pds4_create_xml_index_blackbox.py

```diff
@@ -338,38 +380,154 @@
+            # Compare result to golden copy:
+            # test_files/expected/label_success_2.csv
+            # test_files/expected/label_success_2.xml
+            (
+                str(expected_dir / 'label_success_2.csv'),
+                'generated_label_2.csv', None,
+                [
+                    str(test_files_dir),
+                    str(labels_dir.name / Path('tester_label_1.xml')),
+                    '--generate-label',
+                    'metadata',
                     '--fixed-width',
-                    '--output-index-file'
+                    '--config',
+                    str(expected_dir / 'tester_config.yaml')
+                ]
+            ),
+
+            # Executable command: pds4_create_xml_index ../test_files/labels "tester_label_1.xml" "tester_label_2.xml" "tester_label_3.xml" --limit-xpaths-file ../test_files/samples/element_5.txt --add-extra-file-info filename,filepath,lid,bundle,bundle_lid --generate-label ancillary --config ../test_files/expected/tester_config.yaml --output-index-file generated_label_3.csv
+            # Compare result to golden copy:
+            # test_files/expected/label_success_3.csv
+            # test_files/expected/label_success_3.xml
+            (
+                str(expected_dir / 'label_success_3.csv'),
+                'generated_label_3.csv', None,
+                [
+                    str(test_files_dir),
+                    str(labels_dir.name / Path('tester_label_1.xml')),
+                    str(labels_dir.name / Path('tester_label_2.xml')),
+                    str(labels_dir.name / Path('tester_label_3.xml')),
+                    '--limit-xpaths-file',
+                    str(samples_dir / 'element_5.txt'),
+                    '--add-extra-file-info',
+                    'filename,filepath,lid,bundle,bundle_lid',
+                    '--sort-by',
+                    'filename',
+                    '--generate-label',
+                    'ancillary',
+                    '--config',
+                    str(expected_dir / 'tester_config.yaml')
                 ]
             )
         ]
     )
-def test_success(golden_file, new_file, cmd_line):
+def test_success(golden_file, new_file_index, new_file_headers, cmd_line):
     # Create a temporary directory
     with tempfile.TemporaryDirectory(dir=test_files_dir.parent) as temp_dir:
         temp_dir_path = Path(temp_dir)
 
-        # THE PATH TO THE NEW FILE
-        path_to_file = temp_dir_path / new_file
-        # Call main() function with the simulated command line arguments
-        cmd_line.append(str(path_to_file))
-        tools.main(cmd_line)
+        if new_file_index is None and new_file_headers is None:
+            os.chdir(temp_dir_path)
```

Comment:
Changing the working directory while running a test suite is generally a bad idea because, as a global change, it can affect other tests later. Instead, just use absolute paths for all the filenames, which I think you're mostly doing already anyway.